app/clients/summary_client.py

```
@@ -49,35 +110,92 @@ async def summarize_batch(self, texts: List[str]) -> List[dict]:
-            raise
+            except httpx.TimeoutException as e:
+                last_exception = e
+                if attempt < self.max_attempts:
+                    # Exponential backoff with jitter
+                    delay = min(
+                        self.initial_delay * (2 ** (attempt - 1)),
+                        self.max_delay
+                    )
+                    # Add jitter (±25%)
+                    import random
+                    jitter = delay * 0.25 * (2 * random.random() - 1)
```

Ruff (0.14.8)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)